ChessVariantsLogic/Movement/JumpPattern.cs

```diff
@@ -36,6 +36,44 @@ public JumpPattern(int xOffset, int yOffset)
     /// <returns>-1</returns>
     public int MaxLength {get {return -1; } }
 
+
+    /// <summary>
+    /// Returns a bool indicating wheter this and <paramref name="other"/> are equal.
+    /// </summary>
+    /// <param name="other"> is the other IPattern to be compared to this.</param>
+    /// <returns>true if this is equals to <paramref name="other"/>, otherwise false.</returns>
+    public bool Equals(IPattern? other)
+    {
+        if (other == null) return false;
+        return this.XDir.Equals(other.XDir)
+            && this.YDir.Equals(other.YDir);
+    }
```

Comment:
This would return `true` for a RegularPattern with same `XDir` and `YDir` values but not same `MinLength` and `MaxLength`, right?

Reply:
You are correct. Should be fixed now that the interface `IPattern` has been changed to an abstract class `Pattern` instead :)